client/src/components/WorkflowInvocationState/WorkflowInvocationState.vue

```diff
@@ -219,11 +220,22 @@ async function pollJobStatesUntilTerminal() {
 function onError(e: any) {
     console.error(e);
 }
-function onCancel() {
-    emit("invocation-cancelled");
-}
-function cancelWorkflowSchedulingLocal() {
-    cancelWorkflowScheduling(props.invocationId).then(onCancel).catch(onError);
+async function onCancel() {
+    try {
+        cancellingInvocation.value = true;
+        await cancelWorkflowScheduling(props.invocationId);
+    } catch (e) {
+        onError(e);
+    } finally {
+        emit("invocation-cancelled");
+
+        // Update the invocation state to reflect the cancellation
+        setTimeout(async () => {
```

Comment:
Why the arbitrary timeout here ? I think the responsibility of this method is to to make the request and disable the cancel button. When all of this arrives at the scheduler depends on how busy the workflow scheduler is.

Reply:
I just added the fetches (along with a timeout) because without them, this can happen:

https://github.com/user-attachments/assets/70520d79-45c3-471f-98cd-249a65f19992

I.e.: The invocation state is never updated in the store since we stop fetching.

Reply:
Well the state of those jobs is still going to change though, no fixed amount of time changes that. You can stop polling the invocation, but not the jobs.

Reply:
You're right, and the jobs do poll after this, until `jobStatesTerminal === true`
But the invocation object locally here doesn't update; I'm looking for a way to move the `cancelWorkflowSceduling` function to the invocation store to update invocation there when it is cancelling.

Reply:
Looks good, thank you!